minecraft/franchise/discovery.go

```diff
@@ -0,0 +1,107 @@
+package franchise
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/sandertv/gophertunnel/minecraft/franchise/internal"
+)
+
+// Discover obtains a Discovery for the specific version, which includes environments for various franchise services.
+// It sends a GET request using [http.DefaultClient]. The version is typically [protocol.CurrentVersion] to ensure
+// compatibility with the current version of the protocol package.
+func Discover(build string) (*Discovery, error) {
+	req, err := http.NewRequest(http.MethodGet, discoveryURL.JoinPath(build).String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("make request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", internal.UserAgent)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("%s %s: %w", req.Method, req.URL, err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%s %s: %s", req.Method, req.URL, resp.Status)
+	}
+	var result internal.Result[*Discovery]
+	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+		return nil, fmt.Errorf("decode response body: %w", err)
+	}
+	if result.Data == nil {
+		return nil, errors.New("minecraft/franchise: Discover: result.Data is nil")
+	}
+	return result.Data, nil
+}
+
+// Discovery provides access to environments for various franchise services based on the game
+// version. It can be obtained from Discover using a specific game version.
+//
+// Example usage:
+//
+//	discovery, err := franchise.Discover(protocol.CurrentVersion)
+//	if err != nil {
+//		log.Fatalf("Error obtaining discovery: %s", err)
+//	}
+//
+//	// Look up and decode an environment for authorization.
+//	auth := new(franchise.AuthorizationEnvironment)
+//	if err := discovery.Environment(auth, franchise.EnvironmentTypeProduction); err != nil {
+//		log.Fatalf("Error reading environment for %q: %s", a.EnvironmentName(), err)
+//	}
```

Comment:
Would it be better to have an `ExampleDiscovery()` method (similar to `ExampleDial()` and `ExampleListen()`) so they show correctly in the go docs?